x-pack/plugins/actions/server/builtin_action_types/lib/request_oauth_client_credentials_token.ts

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import qs from 'query-string';
+import axios from 'axios';
+import { Logger } from '../../../../../../src/core/server';
+import { request } from './axios_utils';
+import { ActionsConfigurationUtilities } from '../../actions_config';
+
+export const OAUTH_CLIENT_CREDENTIALS_GRANT_TYPE = 'client_credentials';
+
+interface ClientCredentialsRequestParams {
+  scope?: string;
+  clientId?: string;
+  clientSecret?: string;
+}
+
+export interface ClientCredentialsResponse {
+  tokenType: string;
+  accessToken: string;
+  expiresIn: number;
+}
+
+export async function requestOAuthClientCredentialsToken(
+  tokenUrl: string,
+  logger: Logger,
+  params: ClientCredentialsRequestParams,
+  configurationUtilities: ActionsConfigurationUtilities
+): Promise<ClientCredentialsResponse> {
+  const axiosInstance = axios.create();
+  const { clientId, clientSecret, scope } = params;
+
+  const res = await request({
+    axios: axiosInstance,
+    url: tokenUrl,
+    method: 'post',
+    logger,
+    data: qs.stringify({
+      scope,
+      client_id: clientId,
+      client_secret: clientSecret,
+      grant_type: OAUTH_CLIENT_CREDENTIALS_GRANT_TYPE,
+    }),
+    headers: {
+      'Content-Type': 'application/x-www-form-urlencoded; charset=UTF-8',
```

Comment:
Will OAuth Client Credentials requests always be x-www-form-urlencoded?

Reply:
Yes, according to RFC https://datatracker.ietf.org/doc/html/rfc6749#section-4.4